.github/workflows/update-docker-image.yml

```diff
@@ -0,0 +1,50 @@
+on:
+  push:
+    branches:
+      - dev # add more branches as required
+    # tags:
+    #   - [v*]
+name: Update and Test Docker Image
+jobs:
+  publish:
+    name: Build and publish
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+    env:
+      IMAGE: openmined/pydp
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      # - name: Get the version #
+      #   id: get_version
+      #   run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+      - name: Show python version and build image name  
+        id: build_name
+        run: |
+          python -c "import sys; print(sys.version)"
+          echo "::set-output name=image_name::${IMAGE}-${GITHUB_REF##*/}-${{matrix.python-version}}"
+
+      - name: Build images
+        run: docker build -t ${{steps.build_name.outputs.image_name}} --build-arg python_version=${{matrix.python-version}} -f Dockerfile .
+
+      - name: Tag images
+        run: |
+          docker tag ${{steps.build_name.outputs.image_name}} ${{steps.build_name.outputs.image_name}}:latest
+          docker tag ${{steps.build_name.outputs.image_name}} ${{steps.build_name.outputs.image_name}}:$GITHUB_SHA
+      - name: Run tests
+        run:
+          | # Not sure if this is what is desired. If a test fails the whole build fails - can mark flaky tests if needed?
+          docker run ${{steps.build_name.outputs.image_name}}:latest /bin/bash -c  "pipenv install --dev && pipenv run pytest ./tests"
```

Comment:
This can be a separate workflow

Reply:
I think it would be nicer to run the tests against the docker images before pushing them out the door. 
It won't add much to the total time taken to run the action. And the final shipped artifact will be identical.